From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EvilOlaf <werner@armbian.com>
Date: Sat, 25 Jul 2026 19:30:09 +0200
Subject: [ARCHEOLOGY] cix: add bleedingedge 7.2 support

> X-Git-Archeology: > recovered message: > patches borrowed from https://gitlab.com/ncz-os/meta-cix/-/tree/main/recipes-kernel/linux-cix-sky1-ncz/linux-cix-sky1-ncz-7.2/patches-7.2?ref_type=heads
> X-Git-Archeology: - Revision 50c75ea446ca1c4e723b8c0f6078f24806b71301: https://github.com/armbian/build/commit/50c75ea446ca1c4e723b8c0f6078f24806b71301
> X-Git-Archeology:   Date: Sat, 25 Jul 2026 19:30:09 +0200
> X-Git-Archeology:   From: EvilOlaf <werner@armbian.com>
> X-Git-Archeology:   Subject: cix: add bleedingedge 7.2 support
> X-Git-Archeology:
---
 drivers/regulator/fixed.c | 13 ++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/fixed.c b/drivers/regulator/fixed.c
index 111111111111..222222222222 100644
--- a/drivers/regulator/fixed.c
+++ b/drivers/regulator/fixed.c
@@ -231,8 +231,17 @@ fwnode_get_fixed_voltage_config(struct device *dev,
 		return ERR_PTR(-EINVAL);
 	}
 
-	if (init_data->constraints.boot_on)
-		config->enabled_at_boot = true;
+	/*
+	 * CIX Sky1 ACPI describes fixed rails that firmware has already
+	 * enabled, but does not set regulator-boot-on.  Preserve that state
+	 * when acquiring the enable GPIO.  Requesting it low here glitches
+	 * live PCIe supplies after enumeration; in particular, both RTL8127
+	 * devices disappear between r8169 probe and open on BIOS 1.0.
+	 *
+	 * This restores the behavior of the original CIX ACPI fixed-regulator
+	 * support before the generic fwnode constraint parser replaced it.
+	 */
+	config->enabled_at_boot = true;
 
 	fwnode_property_read_u32(np, "startup-delay-us", &config->startup_delay);
 	fwnode_property_read_u32(np, "off-on-delay-us", &config->off_on_delay);
-- 
Armbian

